pkg/storage/chunk/schema_config.go

```diff
@@ -446,3 +466,34 @@ func (cfg *PeriodicTableConfig) TableFor(t model.Time) string {
 func (cfg *PeriodicTableConfig) tableForPeriod(i int64) string {
 	return cfg.Prefix + strconv.Itoa(int(i))
 }
+
+// Generate the appropriate external key based on cfg.Schema, chunk.Checksum, and chunk.From
+func (cfg SchemaConfig) ExternalKey(chunk Chunk) string {
+	p, err := cfg.SchemaForTime(chunk.From)
+	v, _ := p.VersionAsInt()
```

Comment:
Should we err check this ?

Reply:
I'm inclined to leave this as-is after merging the changes from @owen-d which include a test case for the error condition in `VersionAsInt()`